optuna/integration/sklearn.py

```diff
@@ -94,6 +99,48 @@ def _check_sklearn_availability():
         )
 
 
+# NOTE Original implementation:
+# https://github.com/scikit-learn/scikit-learn/blob/ \
+# 8caa93889f85254fc3ca84caa0a24a1640eebdd1/sklearn/utils/validation.py#L131-L135
+def _is_arraylike(x):
+    # type: (Any) -> bool
+
+    return (
+        hasattr(x, '__len__') or
+        hasattr(x, 'shape') or
+        hasattr(x, '__array__')
+    )
+
+
+# NOTE Original implementation:
+# https://github.com/scikit-learn/scikit-learn/blob/ \
+# 8caa93889f85254fc3ca84caa0a24a1640eebdd1/sklearn/utils/validation.py#L217-L234
+# It removed the check if an input is scipy sparse matrix
```

Comment:
Sorry, but I couldn't get the point of this comment.
When I checked the difference between the original code and this code, I understood that the latter one does not have the conversion from scipy sparse matrix to csr. Could you mention it and add the reason?

Reply:
Sorry for my ambiguous comment...

> the latter one does not have the conversion from scipy sparse matrix to csr.

Yes, you're right.
Actually, I didn't ignore a sparse matrix because I didn't find any use case. 🤕
(In 9c6da5d, I added the support for a sparse matrix.)

Reply:
Sorry for my ambiguous comment...

> the latter one does not have the conversion from scipy sparse matrix to csr.

Yes, you're right.
Actually, I ignored a sparse matrix because I didn't find any use case. 🤕
(In 9c6da5d, I added the support for a sparse matrix.)